astropy/units/quantity_helper/function_helpers.py

```diff
@@ -1147,6 +1176,16 @@ def eig(a, *args, **kwargs):
     return (a.value,) + args, kwargs, (a.unit, dimensionless_unscaled), None
 
 
+if not NUMPY_LT_2_0:
+    # these functions were added in numpy 2.0
+
+    @function_helper(module=np.linalg)
+    def outer(x1, x2, /):
+        # maybe this one can be marked as subclass-safe in the near future ?
+        # see https://github.com/numpy/numpy/pull/25101#discussion_r1419879122
+        return (x1, x2), {}, x1.unit * x2.unit, None
```

Comment:
Thanks for adding the comment. Note that this still will break if either `x1` or `x2` is not a `Quantity` (which is allowed). You'll sadly need to follow the logic in `outer`.

Reply:
Thank you, indeed I forgot to follow up on your first comment. Commit incoming !